src/run.rs

```diff
@@ -214,10 +219,47 @@ pub fn clean_up(
         );
     }
 
+    for subprocess in subprocesses.iter_mut() {
+        #[cfg(unix)]
+        {
+            use std::convert::TryInto;
+
+            unsafe {
+                libc::kill(
+                    subprocess
+                        .id()
+                        .try_into()
+                        .context("cannot get process id")?,
+                    libc::SIGTERM,
+                );
+            }
+        }
+        #[cfg(windows)]
+        {
+            let _ = subprocess.kill();
+            let _ = subprocess.wait();
+        }
+    }
+
+    #[cfg(unix)]
+    {
+        std::thread::sleep(std::time::Duration::from_secs(2));
+
+        for subprocess in subprocesses.iter_mut() {
+            match subprocess.try_wait() {
+                Ok(Some(_)) => {}
+                _ => {
+                    let _ = subprocess.kill();
+                    let _ = subprocess.wait();
+                }
+            }
+        }
+    }
```

Comment:
The logic with the feature flags can be written better. The only difference between unix and windows here is that in unix there is a softkill first.

Both do the hard kill at the end.

Try to improve this code for readability